doc/en/source/k8sIntegration.rst

```diff
@@ -0,0 +1,186 @@
+
+ 
+**Automated Rollouts & Rollback Rollouts** can be achieved by rolling update. Rolling updates are the default strategy to update the running version of your app. It updates cycles previous Pod out and bring newer Pod in incrementally.
+When any introduced change that breaks production, then there should have a plan to roll back that change Kubernetes and kubectl offer a simple mechanism to roll back changes to resources such as Deployments.
+
+
+**Helm** is a tool that streamlines installing and managing Kubernetes applications. It helps in managing Kubernetes applications. Helm Charts helps to define, install, and upgrade even the most complex Kubernetes application.
+FogFlow document would be updated with the functioning details of above features to understand and access the Kubernetes environment well.
+
+
+
+
+.. figure:: figures/k8s-architecture.png
+
+
+
+How Kubernetes component works:
+------------------------------
+The **API Server** is the main management point of the entire cluster. In short, it processes REST operations, validates them, and updates the corresponding objects in etcd. When user interacts with Kubernetes cluster using the kubectl command-line interface, actually the communication establishes with the master API Server component.
+
+**Etcd** is a distributed, consistent key-value store used for configuration management, service discovery, and coordinating distributed work.
+
+Kubernetes **Controller Manager** is a daemon that embeds the core control loops shipped with Kubernetes. Basically, a controller watches the state of the cluster through the API Server watch feature and, when it gets notified, it makes the necessary changes attempting to move the current state towards the desired state.
+
+The **Scheduler** watches for unscheduled pods and binds them to nodes via the /binding pod subresource API, according to the availability of the requested resources, quality of service requirements, affinity and anti-affinity specifications, and other constraints. 
+
+**Kubelet** is an agent that runs on each node in the cluster. It makes sure that containers are running in a Pod. Kubelet is a node component run on every node.
+
+**kube-proxy** is a network proxy that runs on each node in the cluster, it maintains network rules on nodes. These network rules allow network communication to Pods from network sessions inside or outside of your cluster.
+
+
+Here are the prerequisite commands for running FogFlow on K8s:
+
+1. docker
+2. Kubernetes
+3. Helm
+
+.. important:: 
+	**please also allow your user to execute the Docker Command without Sudo**
+	
+To install Kubernetes, please refer to `Install Kubernetes`_,
+To install Helm, please refer to `Install Helm`_,
+
+.. _`Install Kubernetes`: https://medium.com/@vishal.sharma./installing-configuring-kubernetes-cluster-on-ubuntu-18-04-lts-hosts-f37b959c8410
```

Comment:
I think you can give official Kubeadm and K8s documentation link for installation